lib/db-sync-progress.js

```diff
@@ -24,19 +26,26 @@ module.exports = function (db, opts) {
   return stream
 
   function onFeed (feed) {
-    feeds.push(feed)
+    if (feeds.has(feed)) return
+    feeds.set(feed.key, feed)
     feed.ready(updateFeed.bind(null, feed))
-    feed.on('download', onDownload)
-    function onDownload () {
+    feed.on('download', listener)
+    feed.on('upload', listener)
+
+    function listener () {
       updateFeed(feed)
     }
-    listeners.push({ feed: feed, listener: onDownload })
+    listeners.push({ feed, listener })
   }
 
   function updateFeed (feed) {
-    progress[feed.key.toString('hex')] = feed.downloaded(0, feed.length)
-    var total = feeds.reduce(function (acc, feed) { return acc + feed.length }, 0)
-    var sofar = feeds.reduce(function (acc, feed) { return acc + feed.downloaded(0, feed.length) }, 0)
+    var all = Array.from(feeds.values())
+    var total = all.reduce(function (acc, feed) {
+      return acc + feed.length
+    }, 0)
+    var sofar = all.reduce(function (acc, feed) {
+      return acc + feed.stats.totals.downloadedBlocks + feed.stats.totals.uploadedBlocks
```

Comment:
I *think* this might give misleading results?

My understanding of how `stats.totals` works is that it records every upload and download, across multiple peer sessions. We share the same multifeed and thus the same hypercore references across all syncs, so if someone syncs with on peer they'd maybe see their `uploadedBlocks` go to `100`, but on a second sync (/wo restarting Mapeo), they'll see `uploadedBlocks` **start** at `100`.

In the `v9` branch, I include the totals in the handshake, and count 'upload' and 'download' events. Maybe we just continue to track downloads only for now, but do use the 'upload' events for detection of inactivity?

Reply:
sounds good